src/product.cc

```diff
@@ -78,14 +78,14 @@ int Product::package_id() {
 }
 
 void Product::ChangePackageId(int new_package_id) {
-  if (ctx_ != NULL) {
+  if (ctx_ == NULL) {
     throw ValueError("Package Id cannot be changed with NULL context");
   }
```

Comment:
@nuclearkatie - in `material.cc` we just return rather than throwing an error in this case. Do we need to throw an error?